network/src/sync/blocks/batched.rs

```diff
@@ -133,55 +60,25 @@ impl SyncMaster {
         Ok(sent)
     }
 
-    /// receives an arbitrary amount of inbound sync messages with a given timeout.
-    /// if the passed `handler` callback returns `true`, then the loop is terminated early.
-    /// if the sync stream closes, the loop is also terminated early.
-    async fn receive_messages<F: FnMut(SyncInbound) -> bool>(
-        &mut self,
-        timeout_sec: u64,
-        moving_timeout_sec: u64,
-        mut handler: F,
-    ) {
-        let end = Instant::now() + Duration::from_secs(timeout_sec);
-        let mut moving_end = Instant::now() + Duration::from_secs(moving_timeout_sec);
-        loop {
-            let timeout = tokio::time::sleep_until(end.min(moving_end)).fuse();
-            pin_mut!(timeout);
-            select! {
-                msg = self.incoming.recv().fuse() => {
-                    if msg.is_none() {
-                        break;
-                    }
-                    metrics::decrement_gauge!(snarkos_metrics::queues::SYNC_ITEMS, 1.0);
-                    if handler(msg.unwrap()) {
-                        break;
-                    }
-                    moving_end += Duration::from_secs(moving_timeout_sec);
-                },
-                _ = timeout => {
-                    break;
-                }
-            }
-        }
-    }
-
     async fn receive_sync_hashes(&mut self, max_message_count: usize) -> HashMap<SocketAddr, Vec<Digest>> {
         const TIMEOUT: u64 = 5;
         let mut received_block_hashes = HashMap::new();
 
-        self.receive_messages(TIMEOUT, TIMEOUT, |msg| {
-            match msg {
-                SyncInbound::BlockHashes(addr, hashes) => {
-                    received_block_hashes.insert(addr, hashes.into_iter().map(|x| -> Digest { x.0.into() }).collect());
-                }
-                SyncInbound::Block(..) => {
-                    warn!("received sync block prematurely");
+        self.base
+            .receive_messages(TIMEOUT, TIMEOUT, |msg| {
+                match msg {
+                    SyncInbound::BlockHashes(addr, hashes) => {
+                        received_block_hashes
+                            .insert(addr, hashes.into_iter().map(|x| -> Digest { x.0.into() }).collect());
+                    }
+                    SyncInbound::Block(..) => {
+                        warn!("received sync block prematurely");
+                    }
                 }
-            }
-            //todo: fail if peer sends > 1 block hash packet
-            received_block_hashes.len() >= max_message_count
-        })
-        .await;
+                //todo: fail if peer sends > 1 block hash packet
```

Comment:
is this `todo` still applicable?

Reply:
it was always just an idea, and remains a valid idea